### PR TITLE
Shade Guava and hcommon-hive-metastore

### LINKED ISSUE #25: Vacuum tool throws ClassNotFoundException
When running version 2.0.0 the following exception is thrown:

> com.hotels.housekeeping.HousekeepingConfiguration.housekeepingEnvironment(HousekeepingConfiguration.java:68)
>     at com.hotels.housekeeping.HousekeepingConfiguration$$EnhancerBySpringCGLIB$$b919363f.CGLIB$housekeepingEnvironment$0(<generated>)
>     at com.hotels.housekeeping.HousekeepingConfiguration$$EnhancerBySpringCGLIB$$b919363f$$FastClassBySpringCGLIB$$a56e3fa4.invoke(<generated>)
>     at org.springframework.cglib.proxy.MethodProxy.invokeSuper(MethodProxy.java:228)
>     at org.springframework.context.annotation.ConfigurationClassEnhancer$BeanMethodInterceptor.intercept(ConfigurationClassEnhancer.java:355)
>     at com.hotels.housekeeping.HousekeepingConfiguration$$EnhancerBySpringCGLIB$$b919363f.housekeepingEnvironment(<generated>)
>     at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
>     at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
>     at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
>     at java.lang.reflect.Method.invoke(Method.java:498)
>     at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:162)
>     ... 70 more
> Caused by: java.lang.ClassNotFoundException: com.hotels.shaded.com.google.common.collect.ImmutableMap
>     at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
>     at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
>     at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:335)
>     at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
>     ... 81 more

### PR DESCRIPTION
**Pull request by AnanaMJ**
Fixes #25

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -5,9 +5,9 @@
   <parent>
     <groupId>com.hotels</groupId>
     <artifactId>hotels-oss-parent</artifactId>
-    <version>2.0.6</version>
+    <version>2.3.3</version>
```
Comment (massdosage) [inline pom.xml]: Needs a changelog line describing this change

File: housekeeping-vacuum-tool/src/main/assembly/housekeeping-vacuum-tool-package.xml [hunk 2]
```diff
@@ -1,14 +1,26 @@
...
   <id>bin</id>
   <formats>
     <format>tgz</format>
   </formats>
   <baseDirectory>housekeeping-vacuum-tool-${project.version}</baseDirectory>
+  <files>
+    <file>
+      <source>target/housekeeping-vacuum-tool-${project.version}-all.jar</source>
+      <outputDirectory>lib</outputDirectory>
+      <destName>housekeeping-vacuum-tool-all-latest.jar</destName>
```
Comment (massdosage) [inline housekeeping-vacuum-tool/src/main/assembly/housekeeping-vacuum-tool-package.xml]: Need to mention in changelog that this jar now has a new name

File: housekeeping-vacuum-tool/pom.xml [hunk 3]
```diff
@@ -95,6 +95,7 @@
               <artifactSet>
                 <includes>
                   <include>com.hotels:housekeeping-vacuum-tool</include>
+                  <include>com.hotels:hcommon-hive-metastore</include>
```
Comment (massdosage) [inline housekeeping-vacuum-tool/pom.xml]: Need a changelog entry for this, saying that this is now shaded into the tool and removed from the lib folder.

File: pom.xml (hunk 1 above)
Comment (AnanaMJ) [reply pom.xml]: Should I say what version it was changed from?

Comment (massdosage) [reply pom.xml]: Might as well - "Upgraded hotels-oss-parent to  2.3.3 (was 2.0.6)" or something similar.

**Comment by AnanaMJ** [comment]
@massdosage The SNAPSHOT version is 2.1.0 and it was updated because of the metastore tunnelling changes, which removed `MetastoreTunnel`. However, this was not released yet, so can I keep the version 2.1.0-SNAPSHOT?

**Comment by massdosage** [comment]
Ah yes, so we've already incremented the minor version from that previous change, so yeah, keep it like it is.

**Pull request merged**